src/flatdirectory.js

```diff
@@ -246,6 +249,7 @@ export class FlatDirectory {
             totalStorageCost += value;
             // gas cost
             if (gasLimit === 0) {
+                blobHashArr = blobHashArr ? blobHashArr : await this.#getBlobHashes(blobArr);
```

Comment:
It will compute blob hash twice when gasLimit == 0 && chunkIdArr[0] < oldChunkLength

Reply:
If blobHashArr has a value, it will be returned. If there is no value, it will be requested.